Parse/Internal/Object/Subclassing/ObjectSubclassingController.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading;
+
+#if UNITY
+using TypeInfo = System.Type;
+#endif
+
+namespace Parse.Internal {
+  internal class ObjectSubclassingController : IObjectSubclassingController {
+    private ReaderWriterLockSlim mutex;

```

Comment:
curious, any reason we're using `ReaderWriterLockSlim` here instead of using usual `lock(mutex)` other than the performance boost it gives?